mikeio/spatial/FM_geometry.py

```diff
@@ -1097,21 +1110,16 @@ def isel(self, idx=None, axis="elements", keepdims=False):
         --------
         find_index : find element indicies for points or an area
         """
-        if (np.isscalar(idx) or len(idx)) == 1 and (not keepdims):
-            coords = self.element_coordinates[idx].flatten()
 
-            if self.is_layered:
-                return GeometryPoint3D(*coords, projection=self.projection)
-            else:
-                return GeometryPoint2D(coords[0], coords[1], projection=self.projection)
+        if self._type == DfsuFileType.DfsuSpectral1D:
+            return self._nodes_to_geometry(nodes=idx)
         else:
-            if self._type == DfsuFileType.DfsuSpectral1D:
-                return self._nodes_to_geometry(nodes=idx)
-            else:
-                return self.elements_to_geometry(elements=idx, node_layers=None)
+            return self.elements_to_geometry(
+                elements=idx, node_layers=None, keepdims=keepdims
+            )
 
-    def find_index(self, x=None, y=None, coords=None, area=None):
-        """Find element indicies for a number of points or within an area
+    def find_index(self, x=None, y=None, coords=None, area=None) -> Set[int]:
+        """Find a *set* of element indicies for a number of points or within an area
```

Comment:
Maybe state that this means that the returned collection is unordered and unique? (most people would know what a set is, but we also have many in-experienced python users).